src/core/src/collection.rs

```diff
@@ -215,6 +215,14 @@ impl Collection {
         assert_eq!(sig.signatures.len(), 1);
         Ok(sig)
     }
+
+    pub fn intersect_manifest(&self, mf: &Manifest) -> Self {
+        let manifest = self.manifest.intersect_manifest(mf);
+        Self {
+            manifest,
+            storage: self.storage.clone(),
```

Comment:
does just cloning the storage mean that we're carrying around all sigs anyway, but we just don't have access to them via the manifest?

This is probably best since we avoid having to find and copy sigs individually?

Reply:
I think this is really a question for @luizirber at the moment. The alternative would involve references and lifetime annotations, I think.

Reply:
> does just cloning the storage mean that we're carrying around all sigs anyway, but we just don't have access to them via the manifest?
> 
> This is probably best since we avoid having to find and copy sigs individually?

To be clear: I think the answer is yes, esp if it's e.g. a `MemStorage`.

I am worried about duplicating memory unnecessarily. I don't really know how cloning a storage works in practice. I don't like the idea of cloning (which presumably would double the `MemStorage`).

Maybe a better approach here would be to have the result of `Collection::intersect_manifest` take ownership of storage without a clone. I'd need to change the signature of `intersect_manifest` here, will think on't.

Reply:
addressed in [9b9e17e](https://github.com/sourmash-bio/sourmash/pull/3305/commits/9b9e17e791e3d0cd89a254b1701f821bdd460108) and [f4e89f4](https://github.com/sourmash-bio/sourmash/pull/3305/commits/f4e89f4b21c77bc6d68b37f8a4a4f55d6f1c5de9)!

Reply:
`MemStorage` is the odd one, because it does allocate a lot of new memory, but Storage in general doesn't store much (or any) data, mostly metadata to access the actual data.